distributed/executor.py

```diff
@@ -564,11 +564,23 @@ def _gather(self, futures):
         result = pack_data(futures2, data)
         raise gen.Return(result)
 
+    def _threaded_gather(self, qin, qout):
+        """ Internal function for gathering Queue
+        """
+        while True:
+            d = qin.get()
+            f = self.gather(d)
+            qout.put(f)
```

Comment:
At some point in the future we may want to have an internal queue here and depend on the asynchronous side to handle multiple simultaneous gathers.